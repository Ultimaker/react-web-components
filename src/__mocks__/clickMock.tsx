// Copyright (c) 2018 Ultimaker B.V.

export const mockClickEvent = { stopPropagation: jest.fn(), preventDefault: jest.fn() };
