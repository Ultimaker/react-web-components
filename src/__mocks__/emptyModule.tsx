// Copyright (c) 2018 Ultimaker B.V.
module.exports = {};
