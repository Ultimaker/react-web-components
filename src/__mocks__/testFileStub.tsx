// Copyright (c) 2018 Ultimaker B.V.
module.exports = 'test-file-stub';