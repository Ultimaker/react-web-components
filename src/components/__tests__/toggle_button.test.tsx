// Copyright (c) 2018 Ultimaker B.V.
import * as React from 'react';
import { shallow } from 'enzyme';

// component
import ToggleButton from '../toggle_button';

// mocks
import { mockClickEvent } from '../../__mocks__/clickMock';


describe('The Checkbox component', () => {
    let props;
    let wrapper;

    beforeEach(() => {
        props = {
            id: 'testToggleButton',
            value: false,
            onChangeHandler: jest.fn(),
        };
        wrapper = shallow(<ToggleButton {...props} />);
    });

    it('should render', () => {
        expect(wrapper).toMatchSnapshot();
        expect(props.onChangeHandler).not.toHaveBeenCalled();
    });

    it('should not propagate click', () => {
        wrapper.simulate('click', mockClickEvent);
        expect(mockClickEvent.stopPropagation).toBeCalled();
    });
});
