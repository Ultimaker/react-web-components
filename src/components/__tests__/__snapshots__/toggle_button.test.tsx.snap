// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Checkbox component should render 1`] = `
<div
  className="toggle-button-container"
  onClick={[Function]}
>
  <input
    checked={false}
    className="toggle-button"
    id="testToggleButton"
    name="testToggleButton"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    htmlFor="testToggleButton"
  />
</div>
`;
